R/mean-model-workflow.R

```diff
@@ -336,15 +335,18 @@ MeanModelWorkflow <- R6::R6Class(
             pathFolder = self$workflowFolder
           )
           if (self$meanModelSensitivityAnalysis$validateInput()) {
-            dir.create(set$sensitivityAnalysisResultsFolder)
-            # Create the Output of Simulation
+
+            if (!is.null(file.path(self$meanModelSensitivityAnalysis$workflowFolder, self$meanModelSensitivityAnalysis$outputFolder))) {
+              dir.create(file.path(self$meanModelSensitivityAnalysis$workflowFolder, self$meanModelSensitivityAnalysis$outputFolder))
+            }
+
             set$sensitivityAnalysisResultsFileNames <- runSensitivity(
-              simFilePath = file.path(set$inputFilesFolder, paste0(set$simulationSet$simulationName, ".pkml")),
-              variableParameterPaths = self$meanModelSensitivityAnalysis$variableParameterPaths,
-              variationRange = self$meanModelSensitivityAnalysis$variationRange,
-              resultsFileFolder = set$sensitivityAnalysisResultsFolder,
+              simFilePath = set$simulationSet$simulationFile,
+              resultsFileFolder = file.path(self$meanModelSensitivityAnalysis$workflowFolder, self$meanModelSensitivityAnalysis$outputFolder),
               resultsFileName = trimFileName(defaultFileNames$sensitivityAnalysisResultsFile(set$simulationSet$simulationSetName), extension = "csv"),
-              numberOfCores = self$meanModelSensitivityAnalysis$numberOfCores
+              variableParameterPaths = self$meanModelSensitivityAnalysis$settings$variableParameterPaths,

```

Comment:
should not this function take a settings as parameter instead of passing each properties as parameter?

Reply:
Will be done soon